__tests__/Race.test.js

```diff
@@ -1,22 +1,40 @@
-import Race from '../src/domain/Race.js';
-import Car from '../src/domain/Car.js';
+import Race from "../src/domain/Race.js";
+import Car from "../src/domain/Car.js";
 
-describe('자동차 경주 테스트', () => {
-  it('자동차 경주는 기본 5회 진행한다.', () => {
-    const race = new Race([new Car('벤틀리'), new Car('캐딜락')]);
+describe("자동차 경주 테스트", () => {
+  let cars;
 
-    const result = race.start();
+  beforeEach(() => {
+    cars = [new Car("벤틀리"), new Car("캐딜락")];
+  });
+
+  it("자동차 경주 횟수를 정할 수 있다.", () => {
+    const race = new Race(cars, 5);
+
+    const result = race.start(3);
 
     const rounds = result.length;
     expect(rounds).toEqual(5);
   });
 
-  it('자동차 경주 횟수를 변경할 수 있다.', () => {
-    const race = new Race([new Car('벤틀리'), new Car('캐딜락')]);
+  it("자동차 경주 횟수는 0 이상의 숫자를 입력해야 한다.", () => {
+    expect(() => new Race(cars, 0)).toThrow(Race.ROUND_COUNT_ERROR_MESSAGE);
+    expect(() => new Race(cars, "five")).toThrow(
+      Race.ROUND_COUNT_ERROR_MESSAGE
+    );
+  });
 
-    const result = race.start(3);
+  it("경주 마다 자동차 전진 조건을 정할 수 있다", () => {
+    // 모든 자동차가 매 라운드마다 전진하도록 조건을 설정
+    const ROUND_COUNT = 3;
+    const race = new Race(cars, ROUND_COUNT, () => true);
 
-    const rounds = result.length;
-    expect(rounds).toEqual(3);
+    const results = race.start();
+
+    const lastRound = results[results.length - 1];
+    expect(lastRound).toEqual([
+      { name: "벤틀리", location: ROUND_COUNT },
+      { name: "캐딜락", location: ROUND_COUNT },
+    ]);
```

Comment:
```suggestion
    expect(results).toEqual([
      [
        { name: "벤틀리", location: 1 },
        { name: "캐딜락", location: 1 },
      ],
      [
        { name: "벤틀리", location: 2 },
        { name: "캐딜락", location: 2 },
      ],
      [
        { name: "벤틀리", location: 3 },
        { name: "캐딜락", location: 3 },
      ]
    ]);
```
이렇게 만들어야 민지님이 아닌 다른 사람이 코드를 봤을 때 더 이해하기 쉽지 않을까요?

Reply:
지금은 무조건 전진하는데, 새로운 전진조건을 통해 winners가 어떤식으로 산출되는지 보여주는 테스트가 필요할 것 같아요!